packages/ember-glimmer/tests/integration/helpers/partial-test.js

```diff
@@ -73,6 +74,110 @@ moduleFor('Helpers test: {{partial}}', class extends RenderingTest {
     this.assertText('This sub-template is pretty great.');
   }
 
+  ['@test partial using data from {{#each}}']() {
+    this.registerPartial('show-item', '{{item}}');
+
+    this.render(strip`
+      {{#each model.items as |item|}}
+        {{item}}: {{partial 'show-item'}} |
+      {{/each}}`, {
+        model: {
+          items: emberA(['apple', 'orange', 'banana']),
+        }
+      });
+
+    this.assertStableRerender();
+
+    this.assertText('apple: apple |orange: orange |banana: banana |');
+
+    this.runTask(() => this.context.model.items.pushObject('strawberry'));
+
+    this.assertText('apple: apple |orange: orange |banana: banana |strawberry: strawberry |');
+
+    this.runTask(() => set(this.context, 'model', {
+      items: emberA(['apple', 'orange', 'banana']),
+    }));
+
+    this.assertText('apple: apple |orange: orange |banana: banana |');
+  }
+
+  ['@test partial using `{{get` on data from {{#with}}']() {
+    this.registerPartial('show-id', '{{get item "id"}}');
+
+    this.render(strip`
+      {{#with model as |item|}}
+        {{item.name}}: {{partial 'show-id'}}
+      {{/with}}`, {
+        model: { id: 1, name: 'foo' }
+      });
+
+    this.assertStableRerender();
+
+    this.assertText('foo: 1');
+
+    this.runTask(() => set(this.context, 'model.id', 2));
+
+    this.assertText('foo: 2');
+
+    this.runTask(() => set(this.context, 'model.name', 'bar'));
+
+    this.assertText('bar: 2');
+
+    this.runTask(() => set(this.context, 'model', { id: 1, name: 'foo' }));
+
+    this.assertText('foo: 1');
+  }
+
+  ['@test partial using `{{get` on data from {{#each}}']() {
+    this.registerPartial('show-item', '{{get item "id"}}');
+
+    this.render(strip`
+      {{#each items as |item|}}
+        {{item.id}}: {{partial 'show-item'}} |
+      {{/each}}`, {
+        items: emberA([{ id: 1 }, { id: 2 }, { id: 3 }]),
+      });
+
+    this.assertStableRerender();
+
+    this.assertText('1: 1 |2: 2 |3: 3 |');
+
+    this.runTask(() => this.context.items.pushObject({ id: 4 }));
+
+    this.assertText('1: 1 |2: 2 |3: 3 |4: 4 |');
+
+    this.runTask(() => set(this.context, 'items', {
+        items: emberA([{ id: 1 }, { id: 2 }, { id: 3 }]),
+    }));
```

Comment:
This should be:

```
this.runTask(() => set(this.context, 'items',
  emberA([{ id: 1 }, { id: 2 }, { id: 3 }])
));
```

(updated here: https://github.com/emberjs/ember.js/pull/15784)